packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -90,165 +91,255 @@ describe("Presence", () => {
+					});
 				});
 
 				it("is not announced via `attendeeDisconnected` when unknown connection is removed", () => {
 					// Setup
 					presence.events.on("attendeeDisconnected", () => {
-						assert.fail("ateendeeDisconnected should not be emitted for unknown connection.");
+						assert.fail(
+							"`attendeeDisconnected` should not be emitted for unknown connection.",
+						);
 					});
 
-					// Act & Verify - remove unknown connection id
-					presence.removeClientConnectionId("unknownConnectionId");
+					// Act & Verify - remove connection unknown to presence
+					runtime.removeMember("client5");
 				});
 
 				describe("that is joining", () => {
-					it('is announced via `attendeeJoined` with status "Connected" when new', () => {
-						// Setup
-						let newAttendee: ISessionClient | undefined;
+					let newAttendee: ISessionClient | undefined;
+
+					beforeEach(() => {
+						newAttendee = undefined;
 						afterCleanUp.push(
 							presence.events.on("attendeeJoined", (attendee) => {
 								assert(newAttendee === undefined, "Only one attendee should be announced");
 								newAttendee = attendee;
 							}),
 						);
+					});
 
-						// Act - simulate join message from client
-						presence.processSignal("", initialAttendeeSignal, false);
-
-						// Verify
+					function verifyNewAttendee(expectedConnectionId: ClientConnectionId): void {
 						assert(newAttendee !== undefined, "No attendee was announced");
 						assert.equal(
 							newAttendee.sessionId,
-							newAttendeeSessionId,
+							attendeeSessionId,
 							"Attendee has wrong session id",
 						);
 						assert.equal(
 							newAttendee.getConnectionId(),
-							initialAttendeeConnectionId,
+							expectedConnectionId,
 							"Attendee has wrong client connection id",
 						);
 						assert.equal(
 							newAttendee.getConnectionStatus(),
 							SessionClientStatus.Connected,
 							"Attendee connection status is not Connected",
 						);
+					}
+
+					it('first time is announced via `attendeeJoined` with status "Connected"', () => {
+						// Act - simulate join message from client
+						presence.processSignal("", initialAttendeeSignal, false);
+
+						// Verify
+						verifyNewAttendee(initialAttendeeConnectionId);
+					});
+
+					it('second time is announced once via `attendeeJoined` with status "Connected" when prior is unknown', () => {
+						// Setup
+						runtime.removeMember(initialAttendeeConnectionId);
+
+						// Act - simulate join message from client
+						presence.processSignal("", rejoinAttendeeSignal, false);
+
+						// Verify
+						verifyNewAttendee(rejoinAttendeeConnectionId);
+					});
+
+					it.skip('second time is announced once via `attendeeJoined` with status "Connected" when prior is still connected', () => {
+						// Act - simulate join message from client
+						presence.processSignal("", rejoinAttendeeSignal, false);
+
+						// Verify
+						verifyNewAttendee(rejoinAttendeeConnectionId);
+					});
```

Comment:
Struggling to understand this test case, we send a rejoin signal but the initial signal hasn't been sent yet?

Reply:
Consider that the rejoin client (A) had connected at some point. Then they (A) left. Then this client (B) joins. Now the original client (A) rejoins. It (A) will have its original client connection in its history. To (B) this rejoin is the first signal.

Reply:
Ok that makes sense, I think the reason why this test fails is that we are assuming that client A's initial connectionId would remain in audience after it has been disconnected.

Reply:
Similar case for " first time is announced via `attendeeJoined` with status "Connected" even if unknown to audience". If connectinonId unknown to audience, we assume that client is disconnected so we do not announce.

Reply:
Are these tests more for when we don't rely on audience so much for client connectivity information?

Reply:
No. It is unfortunate that the tests need to know that we use audience. Whether we do or do not the test should simulate conditions that runtime will experience. Complicated join/leave/missed-or-delayed messages combinations are real things and we should test for them. Integration tests can let us operate at a higher level without knowing about audience, but we don't have that support yet.